wit/guest.wit

```diff
@@ -3,8 +3,7 @@ interface incoming-handler {
 
     /// Whenever this guest receives a message in one of the subscribed channels, the message is
     /// sent to this handler. The guest is responsible for matching on the channel and handling the
-    /// message accordingly. An Ok result indicates that the message was handled successfully and
-    /// should be considered processed. If explicitly abandoning the message, the guest should return
-    /// error::abandoned with a message indicating why.
+    /// message accordingly. Implementors (such as hosts) calling this interface should make their
+    /// own decisions on how to handle errors returned from this function.
     handle: func(ms: message) -> result<_, error>;
```

Comment:
I still think `handle` should not have a return. Particularly because, like Luke mentioned in the other PR, it sort of creates a lot of questions between the parallels between a `result` from `handle` and a `reply`. Instead, we should just recommend people use `reply` if they'd like to communicate any errors or things of the sort back to the producer instead of back to a Wasm host. Like you mention below, I understand `reply` is different from an HTTP response, but I think mirroring `wasi-http`'s `incoming-handler` here sets a good mental model for how to use the interface.

Reply:
So I can understand the argument for both sides. I still think I'd rather lean towards being able to return an error here. There are two big reasons (in order of importance) for me why this would be important:

1. Some systems may not support (or choose not to support) replying. This means that systems that do this cannot indicate there was an error
2. Some messaging systems might need to translate a failed handler back to their messaging system (i.e. they would need to nack/abandon the message, the same also goes for a success case). Also, let's use the example of composition: What if someone made a `kafka:queue` interface and they had a common component that had retry logic, message flow, etc.? They can then compose that common component with a bunch of "dumb" handler components to create something more complex. This could also go for a custom host implementation of `wasi-messaging` that isn't a component and it has business specific logic around retries. It feels important for an implementor of handle to be able to indicate that the message handling failed

Those two points, especially when taken together, are why I'd prefer to have a result here